analysis/analysis.go

```diff
@@ -38,59 +37,70 @@ func ReceiveRequest(c echo.Context) error {
 	repository := types.Repository{}
 	err := c.Bind(&repository)
 	if err != nil {
-		return c.JSON(http.StatusBadRequest, map[string]string{"result": "error", "details": "Error binding repository."})
+		log.Warning("ReceiveRequest", "ANALYSIS", 101, err)
+		return c.String(http.StatusBadRequest, "This is an invalid JSON.\n")
 	}
 
 	// check-01: is this a git repository URL and a branch?
 	regexpGit := `((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)?`
 	r := regexp.MustCompile(regexpGit)
 	valid, err := regexp.MatchString(regexpGit, repository.URL)
 	if err != nil {
-		return c.JSON(http.StatusBadRequest, map[string]string{"result": "error", "details": "Internal error."})
+		log.Error("ReceiveRequest", "ANALYSIS", 1008, "Repository URL regexp ", err)
+		return c.String(http.StatusInternalServerError, "Internal error 1008.\n")
 	}
 	if !valid {
-		return c.JSON(http.StatusBadRequest, map[string]string{"result": "error", "details": "This is not a valid repository URL."})
+		log.Warning("ReceiveRequest", "ANALYSIS", 102, repository.URL)
```

Comment:
Same here! Warning messages usually indicates that something went not as expected, but does not compromise the progress of the requisition.

Reply:
Same reply as the last one. In this case I am also logging the eventual malicious payload received in `repository.URL`. Example:

`{"repositoryURL": "https://github; echo RCE"}`

Reply:
I have turned these logs into errors and fixed some minor errors in this commit 22c4f68. 😄